pkg/cache/cache.go

```diff
@@ -688,6 +690,52 @@ func (c *Cache) Usage(cqObj *kueue.ClusterQueue) (*ClusterQueueUsageStats, error
 	return stats, nil
 }
 
+type CohortUsageStats struct {
+	WeightedShare int64
+}
+
+func (c *Cache) CohortUsage(cohortObj *kueuealpha.Cohort) (*CohortUsageStats, error) {
+	c.RLock()
+	defer c.RUnlock()
+
+	cohort := c.hm.Cohort(kueue.CohortReference(cohortObj.Name))
+	if cohort == nil {
+		return nil, ErrCohortNotFound
+	}
+
+	stats := &CohortUsageStats{}
+	if c.fairSharingEnabled {
+		weightedShare, _ := dominantResourceShare(cohort, nil)
+		stats.WeightedShare = int64(weightedShare)
+	}
+
+	return stats, nil
+}
+
+func (c *Cache) CohortAncestors(cohortName kueue.CohortReference) []kueue.CohortReference {
+	c.RLock()
+	defer c.RUnlock()
+
+	return c.cohortAncestorsWithoutLock(cohortName)
+}
+
+func (c *Cache) cohortAncestorsWithoutLock(cohortName kueue.CohortReference) []kueue.CohortReference {
+	cohort := c.hm.Cohort(cohortName)
+	if cohort == nil || !cohort.HasParent() {
+		return nil
+	}
+
+	parent := cohort.Parent()
+
+	var ancestors []kueue.CohortReference
+	if !slices.Contains(ancestors, parent.Name) {
+		ancestors = append(ancestors, parent.Name)
+		ancestors = append(ancestors, c.cohortAncestorsWithoutLock(parent.Name)...)
+	}
+
+	return ancestors
+}
```

Comment:
Needs a cycle check. What do you think of this?

```suggestion
func (c *Cache) CohortAncestors(cohortName kueue.CohortReference) ([]kueue.CohortReference, error) {
	c.RLock()
	defer c.RUnlock()
	cohort := c.hm.Cohort(cohortName)
	if cohort == nil {
		return nil, nil
	} else if c.hm.CycleChecker.HasCycle(cohort) {
		return nil, ERR
	}

	var ancestors []kueue.CohortReference
	for cohort != nil {
		ancestors = append(ancestors, cohort.Name)
		cohort = cohort.Parent()
	}
	return ancestors, nil
}
```

Reply:
This is good optimization. Thanks!